Add compatibility to vcpkg buildsystem (#8317)
minetest/minetest: README.md

    VORBIS_LIBRARY                  - Only if building with sound; path to libvorbis.a/libvorbis.so/libvorbis.dll.a
    XXF86VM_LIBRARY                 - Only on Linux; path to libXXf86vm.a/libXXf86vm.so
    ZLIB_DLL                        - Only on Windows; path to zlib1.dll
    ZLIB_INCLUDE_DIR                - Directory that contains zlib.h
    ZLIB_LIBRARY                    - Path to libz.a/libz.so/zlib.lib

### Compiling on Windows

### Requirements

* Visual Studio 2015 or newer https://visualstudio.microsoft.com
* CMake https://cmake.org/download/
* vcpkg https://github.com/Microsoft/vcpkg
* git https://git-scm.com/downloads

### Compiling and Installing the dependencies

> Minetest relays on varios libaries. some are optional and only provide different database backends, and some are mandatory.

Reviewer: This sentence is _somewhat_ redundant, as the `#### a) Using vcpkg to install dependencies` section already lists the optional ones.

Author: ```suggestion
Minetest relays on various third party libraries. 
```
Thanks for your review. 
What do you think about my suggestions?
I does not want to omit the paragraph completely as there is missing a introduction then.

Reviewer: But is there a need for this section at all? It's quite obvious MT depends on various third-party libs. :)

Author: You are right.

> It is highly recommended to use vcpkg as package manager. Of course you can compile the dependencies on your own, but you have to use different buildsystems, solve the dependencies of the dependencies, and may just end up with some linker errors. 

Reviewer: The second sentence is too long and could be omitted IMHO - it just adds to the previous sentence, and doesn't add too much meaning. The README has to be clear and concise. :)

Author: ```suggestion
You can compile the libraries on your own, but it is highly recommended to use vcpkg as package manager.
```

Reviewer: Documentation uses passive voice, and pronouns like "you" and "we" wouldn't sound right. Is this sentence really needed? Again, the README is the official front page of a project, and overly wrong sentences won't make a good impression on the readers.

After you successfully built vcpkg you can easily install the required libaries.
```powershell
vcpkg install irrlicht zlib curl[winssl] openal-soft libvorbis libogg sqlite3 freetype luajit --triplet x64-windows
```
`curl` is optional, but required to read the serverlist, `curl[winssl]` is required to use the content store.

`openal-soft`, `libvorbis` and `libogg` are optional, but required to use sound.

`freetype` is optional, it allows true-type font rendering.

`luajit` is optional and replaces the integrated lua interpreter with a faster just in time interpreter

There are other libaries that are optional, but are not tested if they can build and linked correctly.

`--triplet` specify the target triplet e.g. x64-windows x86-windows

#### b) Compile the dependencies on your own

This is outdated and not recommended. Follow the instructions on https://dev.minetest.net/Build_Win32_Minetest_including_all_required_libraries#VS2012_Build

### Compile Minetest

#### a) Using the vcpkg toolchain and CMakeGUI
1. Start up the CMake GUI
2. Select "Browse Source..." and select DIR/minetest
3. Select "Browse Build..." and select DIR/minetest-build
4. Select "Configure"

> 5. Choose the right visual Studio version and target platform. It have to fit to the target of the installed dependencies

Reviewer: ```suggestion
5. Choose the right visual Studio version and target platform. It has to match the version of the installed dependencies
```

6. Choose "Specify toolchain file for cross-compiling" 
7. Click "Next"
8. Select the vcpkg toolchain file e.g. `D:/vcpkg/scripts/buildsystems/vcpkg.cmake`
9. Click Finish
10. Wait until cmake have generated the cash file
11. If there are any errors, solve them and hit "Configure"
12. Click "Generate"
13. Click "Open Project"
14. Compile Minetest inside Visual studio.

#### b) Using the vcpkg toolchain and the commandline

> Its probably the best, to just create a bat or ps1 file. But you can just paste this two lines inside the powershell window too.

Reviewer: ```suggestion
Run the following script in Powershell:
```

```powershell
cmake . -G"Visual Studio 15 2017 Win64" -DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_BUILD_TYPE=Release -DENABLE_GETTEXT=0 -DENABLE_CURSES=0
cmake --build . --config Release
```

> Make shure that the right Compiler is selected and the path to the vcpkg toolchain file is correct.

Reviewer: ```suggestion
Make sure that the right compiler is selected and the path to the vcpkg toolchain is correct.
```

#### c) Using your own compiled libaries

> This is outdated and not recommended. Follow the instructions on https://dev.minetest.net/Build_Win32_Minetest_including_all_required_libraries#VS2012_Build

Reviewer: ```suggestion
**This is outdated and not recommended**

Follow the instructions on https://dev.minetest.net/Build_Win32_Minetest_including_all_required_libraries#VS2012_Build
```
